### PR TITLE
License file matcher reuse

### PR DESCRIPTION
**Pull request by fviernau**

### CODE REVIEW

File: downloader/src/main/kotlin/vcs/Git.kt [hunk 1]
```diff
@@ -22,7 +22,7 @@ package com.here.ort.downloader.vcs
 import com.here.ort.downloader.WorkingTree
 import com.here.ort.model.VcsInfo
 import com.here.ort.model.VcsType
-import com.here.ort.spdx.LicenseFileMatcher
+import com.here.ort.utils.LicenseFileMatcher
```
Comment (sschuberth) [inline downloader/src/main/kotlin/vcs/Git.kt]: Would you mind adding a rationale to the commit message body?

File: downloader/src/main/kotlin/vcs/Git.kt [hunk 2]
```diff
@@ -65,7 +65,7 @@ class Git : GitBase() {
             val gitInfoDir = File(targetDir, ".git/info").apply { safeMkdirs() }
             val path = vcs.path.let { if (it.startsWith("/")) it else "/$it" }
             File(gitInfoDir, "sparse-checkout").writeText("$path\n" +
-                    LicenseFileMatcher.DEFAULT_NAMES.joinToString("\n") { "/$it" })
+                    LicenseFileMatcher.DEFAULT_MATCHER.fileNames.joinToString("\n") { "/$it" })
```
Comment (sschuberth) [inline downloader/src/main/kotlin/vcs/Git.kt]: This commit effectively reverts a48cc030a7dfe877ac0c6c1185dd8cbc3aee3985. I consider the current code to be cleaner as the code locations where the default file names are used requires less code.

Also, I do not really get the "inconsistencies" argument from the commit message. Could you give a concrete example where having both `DEFAULT_NAMES` and `DEFAULT_MATCHER` could lead to inconsistencies? Both are immutable properties so they should always be in sync.

File: downloader/src/main/kotlin/vcs/Git.kt [hunk 3]
```diff
@@ -22,7 +22,7 @@ package com.here.ort.downloader.vcs
 import com.here.ort.downloader.WorkingTree
 import com.here.ort.model.VcsInfo
 import com.here.ort.model.VcsType
-import com.here.ort.utils.LicenseFileMatcher
+import com.here.ort.utils.FileMatcher
```
Comment (sschuberth) [inline downloader/src/main/kotlin/vcs/Git.kt]: Commit message: How does renaming on its own increase reusability? I guess what you're meaning to say is that a more generic name is more suitable now, as the `DEFAULT_NAMES` property is gone thus the code does  not suggest anymore to be only useful for *license* files.

Anyway, to be hosted this whole commit series did not make much sense to me until this last commit, which finally revealed the real goal: Refactor `LicenseFileMatcher` into a more generic `FileMatcher` that is not necessarily limited to license files. I'd suggest to squash all commits in this PR and use something similar like my last sentence as the commit message title.

File: utils/src/main/kotlin/FileMatcher.kt [hunk 4]
```diff
@@ -24,19 +24,19 @@ import java.nio.file.InvalidPathException
...
+     * The list of file names to consider for matching.
      */
     val fileNames: List<String>
 ) {
     companion object {
         /**
          * A matcher which uses the default license file names.
          */
-        val DEFAULT_MATCHER = LicenseFileMatcher(
+        val DEFAULT_LICENSE_FILE_MATCHER = FileMatcher(
```
Comment (sschuberth) [inline utils/src/main/kotlin/FileMatcher.kt]: How about renaming this to simply `LICENSE_FILE_MATCHER` now, as it's not one of potentially multiple *license* file matchers now anymoer, but *the* license file matcher among the generic file matchers?

File: downloader/src/main/kotlin/vcs/Git.kt (hunk 1 above)
Comment (fviernau) [reply downloader/src/main/kotlin/vcs/Git.kt]: I'm squashing as you suggested.

File: downloader/src/main/kotlin/vcs/Mercurial.kt [hunk 5]
```diff
@@ -23,7 +23,7 @@ import com.here.ort.downloader.VersionControlSystem
 import com.here.ort.downloader.WorkingTree
 import com.here.ort.model.VcsInfo
 import com.here.ort.model.VcsType
-import com.here.ort.spdx.LicenseFileMatcher
+import com.here.ort.utils.LicenseFileMatcher
```
Comment (mnonnenmacher) [inline downloader/src/main/kotlin/vcs/Mercurial.kt]: That entry should be below `CommandLineTool`.

File: utils/src/main/kotlin/LicenseFileMatcher.kt [hunk 6]
```diff
@@ -50,7 +50,8 @@ class LicenseFileMatcher(
         /**
          * A matcher which uses the default license file names.
          */
-        val DEFAULT_MATCHER = LicenseFileMatcher(DEFAULT_NAMES)
+        val DEFAULT_MATCHER =
+            LicenseFileMatcher(DEFAULT_NAMES)
```
Comment (mnonnenmacher) [inline utils/src/main/kotlin/LicenseFileMatcher.kt]: Put this on a single line.

File: downloader/src/main/kotlin/vcs/Git.kt (hunk 2 above)
Comment (fviernau) [reply downloader/src/main/kotlin/vcs/Git.kt]: I have squashed the commits. I'm not sure if you still consider it to be a problem after the squash - thus not addressing it for now.

File: downloader/src/main/kotlin/vcs/Git.kt (hunk 3 above)
Comment (fviernau) [reply downloader/src/main/kotlin/vcs/Git.kt]: Done.

File: utils/src/main/kotlin/FileMatcher.kt (hunk 4 above)
Comment (fviernau) [reply utils/src/main/kotlin/FileMatcher.kt]: Done.

File: downloader/src/main/kotlin/vcs/Mercurial.kt (hunk 5 above)
Comment (fviernau) [reply downloader/src/main/kotlin/vcs/Mercurial.kt]: Done.

File: utils/src/main/kotlin/LicenseFileMatcher.kt (hunk 6 above)
Comment (fviernau) [reply utils/src/main/kotlin/LicenseFileMatcher.kt]: Done.

**Comment by fviernau** [comment]
The only failing test (ClearlyDefinedPackageCurationProviderTest) is unrelated, thus merging regardlessly.

**Pull request merged**